harness/determined/launch/deepspeed.py

```diff
@@ -129,36 +130,52 @@ def create_sshd_cmd() -> List[str]:
     ]
 
 
-def create_deepspeed_env_file() -> None:
-    """Create an env var export file to pass Determined vars to the deepspeed launcher.
+def filter_env_vars(env: Mapping[str, str]) -> Mapping[str, str]:
```

Comment:
why mapping and not dict type here?

Reply:
I was also curious about this

Reply:
Not Ryan, but: `Mapping` is a less strict requirement than `Dict`.  Since he's only using a dict-like interface for the argument, and not the fact that it's literally a `dict`, this supports passing in more things.  For instance, the `__dict__` of a class object is not actually a `dict` but is a `Mapping`.  (Credit StackOverflow)

I think the return argument should be `Dict` though, since we know we're literally returning one, according to a general principle of accepting the most lenient argument type and returning the most strict type.

Reply:
Clarifying: `x.__dict__` isn't a `dict` if x is a class.  So if `A` is a class, `A.__dict__` is a non-dict mapping, whereas `A().__dict__` is an actual `dict`..

Reply:
Ah ok, that makes more sense to me. Maybe using `slots` for the class attributes case or something

Reply:
I'm down with returning a `Dict`.  The reason for `Mapping` is simply because `os.environ` isn't a `Dict`.  Which I found out by failing to pass mypy when I used `Dict`.